app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationEntity.java

```diff
@@ -1,153 +1,58 @@
+import java.lang.annotation.RetentionPolicy;
 
 /**
- * This entity is a "friend" entity from GGMessagingAPI
- * The naming convention is built to enable Gson instance to convert all fields from lower
- * case with underscores to camel case and vice versa.
+ * Created by Gil.Raytan on 10-Jul-16.
  */
-//todo: update all columns and fields and api and...
-public class LocationEntity implements Serializable, DbEntity, ProviderEntity {
-
-    public static final String TABLE_NAME = "GimelGimel";
-
-    public static final Uri CONTENT_URI = formUri(TABLE_NAME, GGContentProvider.authority);
+public class LocationEntity {
 
-    @Override
-    public String tableName() {
-        return TABLE_NAME;
-    }
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({REGULAR, ENEMY, OTHER})
+    public @interface Type {}
+    public static final String REGULAR = "Regular";
+    public static final String ENEMY = "Enemy";
+    public static final String OTHER = "Other";
 
-    @Override
-    public Uri getProviderUri() {
-        return CONTENT_URI;
-    }
+    @SerializedName("location")
+    private PointGeometry mPoint;
 
+    @SerializedName("text")
+    private String mText;
 
-    public static class DB {
-        public static final String ID = "_id";
-        public static final String CREATED_TIMESTAMP = "createdTimestamp";
-        public static final String TITLE = "title";
-        public static final String IS_FAVORITE = "IS_FAVORITE";
-    }
+    @Type
+    @SerializedName("locationType")
+    private String mTypoe;
 
-    public LocationEntity() {
-    }
 
-    /**
-     * Construct the entity using a cursor
-     *
-     * @param cursor the cursor with data, must point to the exact line
-     *               containing the current entity you wish to construct
-     */
-    public LocationEntity(Cursor cursor) {
-        this.id = cursor.getString(cursor.getColumnIndex(DB.ID));
-        this.title = cursor.getString(cursor.getColumnIndex(DB.TITLE));
-        this.createdTimestamp = cursor.getLong(cursor.getColumnIndex(DB.CREATED_TIMESTAMP));
-        this.isFavorite = cursor.getInt(cursor.getColumnIndex(DB.IS_FAVORITE)) != 0;
+    public LocationEntity(PointGeometry point, String text,@Type String type)
+    {
+        this.mPoint = point;
+        this.mText = text;
+        this.mTypoe = type;
     }
 
-    /**
-     * Leading ID of entity, must be unique also from server side
-     */
-    @DbPrimaryKey
-    @DbBinder(dbName = DB.ID)
-    @SerializedName("_id")
-    public String id;
-
-    /**
-     * A unix-timestamp of created time
-     */
-    @DbBinder(dbName = DB.CREATED_TIMESTAMP)
-    public Long createdTimestamp;
-
-    /**
-     * The title of the tip
-     */
-    @DbBinder(dbName = DB.TITLE)
-    public String title;
-
-    /**
-     * Favorite mark, handled locally right now
-     */
-    @DbBinder(dbName = DB.IS_FAVORITE)
-    public boolean isFavorite = false;
+    public PointGeometry getGeometry() {return mPoint;}
```

Comment:
We use geometry for a wider range of geographical object than PointGeometry throughout the project. 
I think it'll be less miss-leading if it was name getPointGeometry()
!fix